formats/json/commonTest/src/kotlinx/serialization/features/JsonClassDiscriminatorTest.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.features
+
+import kotlinx.serialization.*
+import kotlinx.serialization.builtins.*
+import kotlinx.serialization.json.*
+import kotlinx.serialization.modules.*
+import kotlin.test.*
+
+class JsonClassDiscriminatorTest : JsonTestBase() {
```

Comment:
It would be nice to add a test with an example from the documentation 
```
@Serializable
@JsonTypeDiscriminator("class")
abstract class Base
 
@Serializable
@JsonTypeDiscriminator("error_class")
abstract class ErrorClass: Base()

@Serializable
class Message(val object: Base, val error: ErrorClass?)

val message = Json.decodeFromString<Message>("""{"object": {"class":"my.app.BaseMessage", "message": "not found"}, "error": {"error_class":"my.app.GenericError", "error_code": 404}}""")
```

It may also be necessary to check the case of inheritance of the discriminator by abstract classes, like
```
@Serializable
@JsonTypeDiscriminator("class")
abstract class Base
 
@Serializable
abstract class ErrorClass: Base()

@Serializable
class Message(val object: Base, val error: ErrorClass?)

...
```

Reply:
We haven't designed a discriminator inheritance, but we can discuss it if necessary